gopherage/cmd/junit/junit.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package junit
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/tools/cover"
+
+	"k8s.io/test-infra/gopherage/pkg/cov/junit"
+)
+
+type flags struct {
+	OutputFile string
+	Threshold  float32
+}
+
+// MakeCommand returns a `jnuit` command.
+func MakeCommand() *cobra.Command {
+	flags := &flags{}
+	cmd := &cobra.Command{
+		Use:   "junit [profile]",
+		Short: "Summarize coverage profile and produce the result in junit xml format.",
+		Long: `Summarize coverage profile and produce the result in junit xml format.
+Summary done at per-file and per-package level. Any coverage below coverage-threshold will be marked
+with a <failure> tag in the xml produced.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			run(flags, cmd, args)
+		},
+	}
+	cmd.Flags().StringVarP(&flags.OutputFile, "output", "o", "-", "output file")
+	cmd.Flags().Float32VarP(&flags.Threshold, "threshold", "t", .8, "code coverage threshold")
+	return cmd
+}
+
+func run(flags *flags, cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
```

Comment:
if you are always expecting one arg, why not just accept a string than an array?

Reply:
This behavior is consistent with https://github.com/kubernetes/test-infra/blob/master/gopherage/cmd/filter/filter.go.

Reply:
Its not clear from this file. Maybe we need an interface that defines this and all cmd can implement?

Reply:
@Katharine

Reply:
This is method is meeting `cobra.Command.Run`, but adding flags. The pattern lets you capture flag variables scoped to each cobra command instance, it's clearer to keep these args the same because they are passed via simple lambda to the run method.

```go
                Run: func(cmd *cobra.Command, args []string) {
			run(flags, cmd, args)
		},
```